### PR TITLE
(TK-265/TK-268) Remove IP ACE support and weight deny before allow ACEs

### PR DESCRIPTION
**Pull request by camlow325**
Previously, the `ace-compare` function would sort ACEs in an ACL via multiple
criteria - by allow-all, then by exactness, etc.  This PR, for TK-268, changes the
`ace-compare` function to weight all deny ACEs before all allow ACEs,
regardless of any other attributes of the pattern.

Since comparison by `ip` was also removed from the `ace-compare` function,
a number of the ip tests would have needed to be updated in order to still
pass.  Support for `ip` ACEs was already slated to be removed for TK-265
anyway so, rather than writing new code that would have been thrown away
soon anyway, it seemed better to just pull `ip` ACE support out completely as
part of this same PR.  So effectively, this PR satisfies both TK-268 and TK-265.

### CODE REVIEW

File: test/puppetlabs/trapperkeeper/authorization/acl_test.clj [hunk 1]
```diff
@@ -7,139 +7,21 @@
 (use-fixtures :once schema-test/validate-schemas)
 
 (deftest test-compare-entry
-  (let [ip (acl/new-ip :allow "127.0.0.1")
```
Comment (dankreek) [inline test/puppetlabs/trapperkeeper/authorization/acl_test.clj:10]: Mah gawd... the red... it's so beautiful...

File: src/puppetlabs/trapperkeeper/authorization/acl.clj [hunk 2]
```diff
@@ -19,54 +18,28 @@
...
-   * larger pattern length wins
-   * order patterns"
+  "Compare two ACEs.  Deny ACEs always come before allow ACEs.
+
+  For two ACEs of the same type - deny or allow - the 'b' entry always comes
+  before the 'a' entry.  When used as a comparator for a sorted set, this
+  ensures that entries of the same type are ordered first-in, first-out."
   [a :- Entry
    b :- Entry]
-  (or-zero
```
Comment (dankreek) [inline src/puppetlabs/trapperkeeper/authorization/acl.clj:61]: The red, it's just so glorious!

**Comment by dankreek** [comment]
:+1: This all looks good to me.

File: test/puppetlabs/trapperkeeper/authorization/acl_test.clj [hunk 3]
```diff
@@ -198,43 +80,55 @@
...
+    (let [expected-acl [(acl/new-domain :deny "*.domain.com")
+                        (acl/new-domain :deny "my.domain.com")
+                        (acl/new-domain :allow "*.domain.com")
+                        (acl/new-domain :allow "my.domain.com")]]
+      (testing "when allow ACEs added first"
+        (is (= expected-acl (-> (acl/allow "*.domain.com")
+                                (acl/allow "my.domain.com")
+                                (acl/deny "*.domain.com")
+                                (acl/deny "my.domain.com")
+                                ((partial into []))))))
```
Comment (nwolfe) [inline test/puppetlabs/trapperkeeper/authorization/acl_test.clj:96]: Could replace these `partial` lines with just `vec`

**Comment by nwolfe** [comment]
:+1: Love the red

**Pull request merged**